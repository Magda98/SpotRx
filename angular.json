{
	"$schema": "./node_modules/@angular/cli/lib/config/schema.json",
	"version": 1,
	"newProjectRoot": "projects",
	"projects": {
		"spotrx": {
			"projectType": "application",
			"schematics": {
				"@schematics/angular:component": {
					"style": "scss"
				},
				"@schematics/angular:application": {
					"strict": true
				}
			},
			"root": "",
			"sourceRoot": "src",
			"prefix": "app",
			"architect": {
				"build": {
					"builder": "@angular-devkit/build-angular:application",
					"options": {
						"outputPath": "dist/spotrx",
						"index": "src/index.html",
						"browser": "src/main.ts",
						"tsConfig": "tsconfig.app.json",
						"inlineStyleLanguage": "scss",
						"assets": ["src/favicon.ico", "src/assets"],
						"styles": ["src/styles.scss", "src/styles/m3-theme.scss"]
					},
					"configurations": {
						"production": {
							"budgets": [
								{
									"type": "initial",
									"maximumWarning": "1.1mb",
									"maximumError": "2mb"
								},
								{
									"type": "anyComponentStyle",
									"maximumWarning": "2kb",
									"maximumError": "4kb"
								}
							],
							"fileReplacements": [
								{
									"replace": "src/environments/environment.ts",
									"with": "src/environments/environment.prod.ts"
								}
							],
							"outputHashing": "all"
						},
						"development": {
							"optimization": false,
							"extractLicenses": false,
							"sourceMap": true
						}
					},
					"defaultConfiguration": "production"
				},
				"serve": {
					"builder": "@angular-devkit/build-angular:dev-server",
					"configurations": {
						"production": {
							"buildTarget": "spotrx:build:production"
						},
						"development": {
							"buildTarget": "spotrx:build:development"
						}
					},
					"defaultConfiguration": "development"
				},
				"extract-i18n": {
					"builder": "@angular-devkit/build-angular:extract-i18n",
					"options": {
						"buildTarget": "spotrx:build"
					}
				},
				"deploy": {
					"builder": "angular-cli-ghpages:deploy"
				},
				"storybook": {
					"builder": "@storybook/angular:start-storybook",
					"options": {
						"configDir": ".storybook",
						"browserTarget": "spotrx:build",
						"styles": ["src/styles.scss", "src/styles/m3-theme.scss"],
						"compodoc": false,
						"compodocArgs": ["-e", "json", "-d", "."],
						"port": 6006
					}
				},
				"build-storybook": {
					"builder": "@storybook/angular:build-storybook",
					"options": {
						"configDir": ".storybook",
						"browserTarget": "spotrx:build",
						"styles": ["src/styles.scss", "src/styles/m3-theme.scss"],
						"compodoc": false,
						"compodocArgs": ["-e", "json", "-d", "."],
						"outputDir": "storybook-static"
					}
				},
				"lint": {
					"builder": "@angular-eslint/builder:lint",
					"options": {
						"lintFilePatterns": ["src/**/*.ts", "src/**/*.html"]
					}
				}
			}
		}
	},
	"cli": {
		"analytics": false,
		"schematicCollections": ["@angular-eslint/schematics"]
	}
}
