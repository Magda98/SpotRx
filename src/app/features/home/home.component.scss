.home {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  .track-list {
    width: 100%;
  }

  .title {
    margin-bottom: 32px;
  }

  .playlist {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-start;
    padding: 16px 20px;
    border-radius: 10px;
    transition: background-color 0.3s;
    cursor: pointer;
    text-decoration: none;

    &:hover {
      background-color: #2c2f31;
    }
  }

  .playlist-img {
    display: block;
    max-width: 125px;
    max-height: 125px;
    height: 125px;
    object-fit: cover;
    border-radius: 10px;
    margin-bottom: 10px;
    overflow: hidden;

    @media screen and (min-width: 980px) {
      max-width: 200px;
      max-height: 200px;
      height: 200px;
    }
  }

  .playlist-title {
    margin-bottom: 0;
    color: #fff;
    text-decoration: none;
    text-align: center;
    max-width: 125px;
    overflow: hidden;

    @media screen and (min-width: 980px) {
      max-width: 200px;
    }
  }

  .playlist-list {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-template-rows: auto;
    gap: 16px;

    @media screen and (min-width: 980px) {
      grid-template-columns: 1fr 1fr;
    }

    @media screen and (min-width: 1280px) {
      grid-template-columns: 1fr 1fr 1fr 1fr;
    }
  }
}
