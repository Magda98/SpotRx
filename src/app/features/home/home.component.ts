import { Component, inject } from '@angular/core';
import { injectQuery } from '@tanstack/angular-query-experimental';
import { CommonModule, NgOptimizedImage } from '@angular/common';
import { RouterModule } from '@angular/router';
import { NgxSkeletonLoaderModule } from 'ngx-skeleton-loader';
import { TrackService } from '@app/shared/services/track.service';

@Component({
  selector: 'app-home',
  templateUrl: './home.component.html',
  styleUrls: ['./home.component.scss'],
  standalone: true,
  imports: [
    RouterModule,
    CommonModule,
    NgOptimizedImage,
    NgxSkeletonLoaderModule,
  ],
})
export class HomeComponent {
  private trackService = inject(TrackService);
  public featuredPlaylists = injectQuery(() =>
    this.trackService.getFeaturedPlaylists(),
  );
  readonly skeletonLoadingArray = Array.from({ length: 12 }, () => null);
}
