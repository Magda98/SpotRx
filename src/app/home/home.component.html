<div class="home">
    <h1 class="title">Home ðŸ˜Ž</h1>

    <div class="playlist-list">
        <div class="playlist" *ngFor="let item of trackService.getFeaturedPlaylists() | async">
            <img [src]="item.images[0].url">
            <h2>{{item.name}}</h2>
        </div>
    </div>
</div>