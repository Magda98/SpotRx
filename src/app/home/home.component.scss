.home {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  .track-list {
    width: 100%;
  }

  .title {
    margin-bottom: 32px;
  }

  .playlist {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    padding: 16px 20px;
    border-radius: 10px;
    transition: background-color 0.3s;
    cursor: pointer;
    text-decoration: none;

    &:hover {
      background-color: #2c2f31;
    }
  }

  .playlist-img {
    max-width: 200px;
    height: 200px;
    object-fit: cover;
    border-radius: 10px;
    margin-bottom: 10px;
  }

  .playlist-title {
    margin-bottom: 0;
    color: #fff;
    text-decoration: none;
  }

  .playlist-list {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-template-rows: auto;
    gap: 50px;
  }
}