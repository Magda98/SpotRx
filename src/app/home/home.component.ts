import { TrackService } from '../shared/services/track.service';
import { Component, inject } from '@angular/core';
import { injectQuery } from '@tanstack/angular-query-experimental';
import { CommonModule, NgOptimizedImage } from '@angular/common';
import { RouterModule } from '@angular/router';
import { NgxSkeletonLoaderModule } from 'ngx-skeleton-loader';

@Component({
  selector: 'app-home',
  templateUrl: './home.component.html',
  styleUrls: ['./home.component.scss'],
  standalone: true,
  imports: [
    RouterModule,
    CommonModule,
    NgOptimizedImage,
    NgxSkeletonLoaderModule,
  ],
})
export class HomeComponent {
  private trackService = inject(TrackService);
  public featuredPlaylists = injectQuery(() =>
    this.trackService.getFeaturedPlaylists()
  );
  readonly skeletonLoadingArray = Array.from({ length: 12 }, () => null);
}
