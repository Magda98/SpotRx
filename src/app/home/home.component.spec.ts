import { render } from '@testing-library/angular';
import { HomeComponent } from './home.component';
import { RouterModule } from '@angular/router';
import { CommonModule, NgOptimizedImage } from '@angular/common';
import { NgxSkeletonLoaderModule } from 'ngx-skeleton-loader';
import { HttpClientTestingModule } from '@angular/common/http/testing';
import {
  QueryClient,
  provideAngularQuery,
} from '@tanstack/angular-query-experimental';
import { screen } from '@testing-library/angular';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { MockBackendInterceptor } from '../../tests/mock-backend.interceptor';

describe('HomeComponent', () => {
  const renderComponent = () =>
    render(HomeComponent, {
      componentImports: [
        RouterModule,
        CommonModule,
        NgOptimizedImage,
        NgxSkeletonLoaderModule,
      ],
      imports: [HttpClientTestingModule],
      providers: [
        provideAngularQuery(new QueryClient()),
        {
          provide: HTTP_INTERCEPTORS,
          useClass: MockBackendInterceptor,
          multi: true,
        },
      ],
    });

  test('should render', async () => {
    const component = await renderComponent();
    expect(component.container).toMatchSnapshot();
  });

  test('should render title', async () => {
    await renderComponent();
    expect(
      await screen.findByRole('heading', { name: /Home/i })
    ).toBeInTheDocument();
  });
});
