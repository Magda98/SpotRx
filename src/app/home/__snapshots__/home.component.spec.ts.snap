// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should render 1`] = `
<div
  id="root0"
>
  <div
    class="home"
  >
    <h1
      class="title"
    >
      Home ðŸ˜Ž
    </h1>
    
    <div
      class="playlist-list"
    >
      <div
        class="playlist"
      >
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-img"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 200px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-title"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 24px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
      </div>
      <div
        class="playlist"
      >
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-img"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 200px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-title"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 24px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
      </div>
      <div
        class="playlist"
      >
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-img"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 200px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-title"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 24px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
      </div>
      <div
        class="playlist"
      >
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-img"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 200px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-title"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 24px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
      </div>
      <div
        class="playlist"
      >
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-img"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 200px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-title"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 24px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
      </div>
      <div
        class="playlist"
      >
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-img"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 200px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-title"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 24px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
      </div>
      <div
        class="playlist"
      >
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-img"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 200px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-title"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 24px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
      </div>
      <div
        class="playlist"
      >
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-img"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 200px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-title"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 24px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
      </div>
      <div
        class="playlist"
      >
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-img"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 200px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-title"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 24px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
      </div>
      <div
        class="playlist"
      >
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-img"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 200px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-title"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 24px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
      </div>
      <div
        class="playlist"
      >
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-img"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 200px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-title"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 24px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
      </div>
      <div
        class="playlist"
      >
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-img"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 200px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
        <ngx-skeleton-loader
          animation="progress-dark"
          class="playlist-title"
          count="1"
        >
          <div
            aria-busy="true"
            aria-label="loading"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuetext="Loading..."
            class="skeleton-loader progress-dark"
            role="progressbar"
            style="height: 24px; width: 200px; background-color: rgb(44, 47, 49);"
            tabindex="-1"
          >
            
          </div>
          
        </ngx-skeleton-loader>
      </div>
      
    </div>
    
  </div>
</div>
`;
